pkg/smokescreen/smokescreen.go

```diff
@@ -123,21 +123,21 @@ func classifyAddr(config *Config, addr *net.TCPAddr) ipType {
 	}
 }
 
-func safeResolve(config *Config, network, addr string) (*net.TCPAddr, error) {
+func safeResolve(config *Config, network, addr string) (*net.TCPAddr, string, error) {
 	config.StatsdClient.Incr("resolver.attempts_total", []string{}, 1)
 	resolved, err := net.ResolveTCPAddr(network, addr)
 	if err != nil {
 		config.StatsdClient.Incr("resolver.errors_total", []string{}, 1)
-		return nil, err
+		return nil, "", err
 	}
 
 	classification := classifyAddr(config, resolved)
 	config.StatsdClient.Incr(classification.statsdString(), []string{}, 1)
 
 	if classification.IsAllowed() {
-		return resolved, nil
+		return resolved, classification.String(), nil
 	}
-	return nil, denyError{fmt.Errorf("The destination address (%s) was denied by rule '%s'", resolved.IP, classification)}
+	return nil, "", denyError{fmt.Errorf("The destination address (%s) was denied by rule '%s'", resolved.IP, classification)}
```

Comment:
If `!classification.IsAllowed()`, the egress will be blocked. In that case, we want to populate the decision reason with something similar to the error message supplied here, right? Because we want to be able to say something like "Denied because resolved IP matched rule X"

Reply:
Ah, it was my assumption that we'll log out errors when they occur. Is that wrong?

Seems like currently blank strings get set for reason when an error is detected.

Reply:
We do log out errors, but it's still worth making the `decision_reason` accurate.
I don't think `decision_reason` is cleared when an error is present, but maybe it is. In that case, we should maybe supply `error encountered, see error field` or something as the `decision_reason`.

I think it's probably a blank string because in the cases where we supply errors, we're not in the correct context to modify/provide a `decision_reason`.